index.js

```diff
@@ -284,3 +284,27 @@ export function escapePath(path) {
 
 	return path.replace(/[\\.[]/g, '\\$&');
 }
+
+function * deepKeysIterator(object, currentPath) {
+	for (const [key, value] of Object.entries(object)) {
+		const property = typeof currentPath === 'undefined' ? escapePath(key) : `${currentPath}.${escapePath(key)}`;
+
+		if (isObject(value) && !Array.isArray(object)) {
+			yield * deepKeysIterator(value, property);
+		} else {
+			yield property;
+		}
+	}
+}
+
+export function deepKeys(object) {
+	return {
+		* [Symbol.iterator]() {
+			if (!isObject(object) || Array.isArray(object)) {
```

Comment:
I think if you switch to an array path, you should be able to reshuffle things a bit so that you don’t have to duplicate the isObject/isArray checks. Ideally this whole function could/should look like:

```js
export function deepKeys(object) {
  return [...deepKeysIterator(object)]
}
```

If it worked that way, a bonus would be that for the few users who actually _prefer_ an iterator over an array for perf reasons or whatever, could use `deepKeysIterator` directly.